ops/terraform/modules/reporting/reporting.tf

```diff
@@ -0,0 +1,99 @@
+// Reporting module to be used in each AoU deployment (environment). The local "deployment"
+// shares a project with "test", so it's not 1:1.
+
+// Module-specific variables here.
+variable reporting_dataset_id {
+  description = "BigQuery dataset for workbench reporting data."
+  type        = string
+}
+
+locals {
+  //  The module path is in a hidden directory under the running directory of the calling module, but
+  // our view files and schemas for BigQuery are in whatever directory this file is located, and apparently
+  // don't get loaded into that dir by default.
+  // FIXME(jaycarlton): package these files as what Java would call resources, or otherwise find the
+  //  right way to do it. path.module and path. At the vary least, it should work for other user names.
+  //  There has to be a way to separate these cleanly; otherewise you'd have the constraint that everything
+  //  lives in one repo.
+
+  schemas_dir = "/Users/jaycarlton/repos/workbench/ops/terraform/modules/reporting/schemas"
+  views_dir   = "/Users/jaycarlton/repos/workbench/ops/terraform/modules/reporting/views"
+  table_inputs = [
+    {
+      table_id    = "cohort"
+      description = "Workbench Cohorts"
+    },
+    {
+      table_id    = "user"
+      description = "Workbench Users"
+    },
+    {
+      table_id    = "institution"
+      description = "Workbench Verified institutions"
+    },
+    {
+      table_id    = "workspace"
+      description = "Workbench Workspaces (Active Only)"
+    }
+  ]
+
+  table_ids          = [for t in local.table_inputs : t.table_id]
+  table_descriptions = [for t in local.table_inputs : t.description]
+  view_names         = ["latest_users", "latest_cohorts", "latest_workspaces", "latest_institutions", "table_count_vs_time"]
+}
+
+module "reporting" {
+  source     = "terraform-google-modules/bigquery/google"
+  version    = "~> 4.3"
+  dataset_id = var.reporting_dataset_id
+  project_id = var.project_id
+  location   = "US"
+  tables = [for t in local.table_inputs :
+    {
+      table_id    = t.table_id
+      description = t.description
+
+      // The Terraform IDE plugin tells  me  table_id is an invalid reference here, but somehow
+      // it works. Probably something to do with loop variable validation.
+      schema            = "${local.schemas_dir}/${(t.table_id)}.json"
+      time_partitioning = null
+      expiration_time   = null
+      clustering        = []
+      labels = {
+        terraform_managed = "true"
+      }
+    }
+  ]
+
+  views = [for view_name in local.view_names : {
```

Comment:
Do you need a dependency from views -> tables? Or does it let you create a view which references a non-existent table? If so, you can consider using `depends_on`, example: https://github.com/broadinstitute/terraform-terra/blob/master/profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/ingress-bridges.tf#L29

Reply:
Interesting. At the moment, I believe the provider lets BigQuery validates that the tables are real at query time, and I don't believe we want Terraform to inject its opinion. It would be good for sanity checks, but might mean we can't use the loop without a map variable, so it's not clearer.

From experience, if you remove a table that a view needs, you don't find out until you query the view. It's just a glorified saved query.